### ISSUE TITLE
IDEA always reports errors but does not affect compilation

### ISSUE THREAD

**Issue by xiao-cao-x** [Kotlin/kotlinx.serialization#1649]
**Describe the bug**

In my IDEA, kotlinx.serialization is always reporting errors, but it (error reporting) does not affect compilation.

![image](https://user-images.githubusercontent.com/76673990/131220335-65c6b0fa-afff-43e5-a942-10b73267705d.png)

![image](https://user-images.githubusercontent.com/76673990/131220341-91a96fa9-802b-4273-b0b8-fbbdd87a10c2.png)

![image](https://user-images.githubusercontent.com/76673990/131220346-36b60ec7-aef4-4e62-a558-cbc90caf97f4.png)

![image](https://user-images.githubusercontent.com/76673990/131220353-38076915-9feb-4c3f-832a-9288c3b4c18a.png)

**To Reproduce**
Attach a code snippet or test data if possible.

This problem may arise after I added Git to the project.

```
val ktorVersion: String by project
dependencies {
    implementation("io.ktor:ktor-client-core:$ktorVersion")
    implementation("io.ktor:ktor-client-serialization:$ktorVersion")
    implementation("io.ktor:ktor-client-okhttp:$ktorVersion")
}

val compileKotlin: KotlinCompile by tasks
compileKotlin.kotlinOptions {
    jvmTarget = "16"
}
val compileTestKotlin: KotlinCompile by tasks
compileTestKotlin.kotlinOptions {
    jvmTarget = "16"
}
```

**Expected behavior**

It should not report an error, because it does not affect compilation.

**Environment**
 - Kotlin version: [1.5.30]
 - Library version: [io.ktor:ktor-client-serialization:1.6.3]
 - Kotlin platforms: [JVM 16]
 - Gradle version: [7.2]
 - IDE version (if bug is related to the IDE) [IntellijIDEA 2021.2.1]

---
From Google Translate

**Comment by sandwwraith** [comment]
It looks like a Ktor problem, not a kotlinx.serialization one. Or the problem with your setup. try to add `kotlinx-serialization-json` to your dependencies. cc @e5l

**Comment by BenWoodworth** [comment]
I'm pretty sure it's from this bug with the Kotlin 1.5.30 IDEA plugin: [KTIJ-19504](https://youtrack.jetbrains.com/issue/KTIJ-19504) (which will be fixed in 1.5.31)
